include/swift/SIL/GraphOperationBuilder.h

```diff
@@ -55,6 +56,8 @@ class GraphOperationBuilder {
   GraphOperationAttribute &addAttribute(
       const GraphOperationAttribute &attribute);
 
+  ArrayRef<GraphOperationAttribute> getAttributes() const { return Attributes; }
```

Comment:
We discussed a bit in person, and here's a summary:
* If the compiler can const-evaluate the type attribute, then this PR will set the device correctly.
* If the compiler cannot const-evaluate the type attribute, then we'll fall back to CPU for now. (@pschuh will check to make sure that it actually falls back in this case rather than failing). This might not happen very frequently, because dtype should usually be pretty easy to const-evaluate.
* If/when we notice that it's falling back a lot, we can add more complicated stuff that works correctly with nonconstant dtypes: do device assignment at runtime.

Reply:
SGTM!

How can we track/notice if it's falling back a lot? (We can take the discussion outside this PR)

Reply:
The tensorflow code internally verifies that the op has all the necessary attributes in the kernel selection code I'm using to check availability. Thus, it'll bad-status out.

We could parse this bad status and use that to log warning statements to know if it is falling back.